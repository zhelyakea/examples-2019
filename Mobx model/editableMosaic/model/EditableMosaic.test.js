import { onSnapshot } from 'mobx-state-tree';

import EditableMosaic from './index';

const editableMosaicState = {
  isConfirmRemoveLayer: false,
  isConfirmRemovePositions: false,
  dimensions: [1, 2, 3, 4, 5, 6, 8, 9, 10, 12],
  selectedPosition: '',
  mosaic: {
    id: '1000',
    layers: [
      {
        dimension: 3,
        positions: []
      }
    ]
  }
};
const mosaicState = {
  id: '1',
  selectedLayer: 2,
  lowLayersVisibility: true,
  layers: [
    {
      dimension: 4,
      positions: [
        {
          i: '4',
          x: 0,
          y: 1,
          w: 2,
          h: 2,
          role: 0
        },
        {
          i: '3',
          x: 2,
          y: 1,
          w: 2,
          h: 2,
          role: 0
        }
      ]
    },
    {
      dimension: 4,
      positions: [
        {
          i: '2',
          x: 0,
          y: 1,
          w: 2,
          h: 2,
          role: 0
        },
        {
          i: '1',
          x: 2,
          y: 1,
          w: 2,
          h: 2,
          role: 0
        }
      ]
    },
    {
      dimension: 4,
      positions: [
        {
          i: '6',
          x: 0,
          y: 1,
          w: 2,
          h: 2,
          role: 0
        },
        {
          i: '5',
          x: 2,
          y: 1,
          w: 2,
          h: 2,
          role: 0
        }
      ]
    }
  ]
};

it('check setMosaic', () => {
  const editableMosaic = EditableMosaic.create(editableMosaicState);

  onSnapshot(editableMosaic, snapshot => {
    expect(snapshot).toMatchSnapshot();
  });
  editableMosaic.setMosaic(mosaicState);
});
it('check selectPosition', () => {
  const editableMosaic = EditableMosaic.create(editableMosaicState);

  onSnapshot(editableMosaic, snapshot => {
    expect(snapshot).toMatchSnapshot();
  });
  editableMosaic.selectPosition('1');
  expect(editableMosaic.selectedPosition).toBe('1');
});
it('check removeLayerPositions', () => {
  const editableMosaicWithPresetState = { ...editableMosaicState, mosaic: mosaicState };
  const editableMosaic = EditableMosaic.create(editableMosaicWithPresetState);

  onSnapshot(editableMosaic, snapshot => {
    expect(snapshot).toMatchSnapshot();
  });
  editableMosaic.removeLayerPositions();
  const {
    mosaic: { layers, selectedLayer }
  } = editableMosaic;
  const { positions } = layers[selectedLayer];

  expect(positions.length).toBe(0);
});
it('check removeLayer', () => {
  const editableMosaicWithPresetState = { ...editableMosaicState, mosaic: mosaicState };
  const editableMosaic = EditableMosaic.create(editableMosaicWithPresetState);

  onSnapshot(editableMosaic, snapshot => {
    expect(snapshot).toMatchSnapshot();
  });
  editableMosaic.removeLayer(1);
  const mosaic = editableMosaic.mosaic;

  expect(mosaic.layers.length).toBe(2);
  expect(mosaic.selectedLayer).toBe(1);
});
it('check openConfirmRemoveLayer', () => {
  const editableMosaic = EditableMosaic.create(editableMosaicState);

  editableMosaic.openConfirmRemoveLayer(1);

  expect(editableMosaic.indexLayerToRemove).toBe(1);
  expect(editableMosaic.isConfirmRemoveLayer).toBe(true);
});
it('check closeConfirmRemoveLayer', () => {
  const editableMosaic = EditableMosaic.create(editableMosaicState);

  editableMosaic.closeConfirmRemoveLayer();

  expect(editableMosaic.isConfirmRemoveLayer).toBe(false);
});
it('check confirmRemoveLayer', () => {
  const editableMosaicWithPresetState = {
    ...editableMosaicState,
    indexLayerToRemove: 1,
    mosaic: mosaicState
  };
  const editableMosaic = EditableMosaic.create(editableMosaicWithPresetState);

  onSnapshot(editableMosaic, snapshot => {
    expect(snapshot).toMatchSnapshot();
  });
  editableMosaic.confirmRemoveLayer();

  expect(editableMosaic.mosaic.layers.length).toBe(2);
  expect(editableMosaic.isConfirmRemoveLayer).toBe(false);
});

it('check openConfirmRemovePositions', () => {
  const editableMosaic = EditableMosaic.create(editableMosaicState);

  editableMosaic.openConfirmRemovePositions();

  expect(editableMosaic.isConfirmRemovePositions).toBe(true);
});
it('check closeConfirmRemovePositions', () => {
  const editableMosaic = EditableMosaic.create(editableMosaicState);

  editableMosaic.closeConfirmRemovePositions();

  expect(editableMosaic.isConfirmRemovePositions).toBe(false);
});
it('check confirmRemovePositions', () => {
  const editableMosaicWithPresetState = { ...editableMosaicState, mosaic: mosaicState };
  const editableMosaic = EditableMosaic.create(editableMosaicWithPresetState);

  onSnapshot(editableMosaic, snapshot => {
    expect(snapshot).toMatchSnapshot();
  });
  editableMosaic.confirmRemovePositions();
  const {
    mosaic: { layers, selectedLayer }
  } = editableMosaic;
  const { positions } = layers[selectedLayer];

  expect(positions.length).toBe(0);
});
