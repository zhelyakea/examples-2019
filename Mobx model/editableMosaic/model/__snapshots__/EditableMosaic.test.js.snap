// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check confirmRemoveLayer 1`] = `
Object {
  "dimensions": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    9,
    10,
    12,
  ],
  "indexLayerToRemove": 1,
  "isConfirmRemoveLayer": false,
  "isConfirmRemovePositions": false,
  "mosaic": Object {
    "id": "1",
    "layers": Array [
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "1",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "2",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "3",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "4",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
    ],
    "lowLayersVisibility": true,
    "selectedLayer": 1,
  },
  "selectedPosition": "",
}
`;

exports[`check confirmRemovePositions 1`] = `
Object {
  "dimensions": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    9,
    10,
    12,
  ],
  "indexLayerToRemove": undefined,
  "isConfirmRemoveLayer": false,
  "isConfirmRemovePositions": false,
  "mosaic": Object {
    "id": "1",
    "layers": Array [
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "3",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "4",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "1",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "2",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [],
        "visible": true,
      },
    ],
    "lowLayersVisibility": true,
    "selectedLayer": 2,
  },
  "selectedPosition": "",
}
`;

exports[`check removeLayer 1`] = `
Object {
  "dimensions": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    9,
    10,
    12,
  ],
  "indexLayerToRemove": undefined,
  "isConfirmRemoveLayer": false,
  "isConfirmRemovePositions": false,
  "mosaic": Object {
    "id": "1",
    "layers": Array [
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "1",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "2",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "3",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "4",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
    ],
    "lowLayersVisibility": true,
    "selectedLayer": 1,
  },
  "selectedPosition": "",
}
`;

exports[`check removeLayerPositions 1`] = `
Object {
  "dimensions": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    9,
    10,
    12,
  ],
  "indexLayerToRemove": undefined,
  "isConfirmRemoveLayer": false,
  "isConfirmRemovePositions": false,
  "mosaic": Object {
    "id": "1",
    "layers": Array [
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "3",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "4",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "1",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "2",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [],
        "visible": true,
      },
    ],
    "lowLayersVisibility": true,
    "selectedLayer": 2,
  },
  "selectedPosition": "",
}
`;

exports[`check selectPosition 1`] = `
Object {
  "dimensions": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    9,
    10,
    12,
  ],
  "indexLayerToRemove": undefined,
  "isConfirmRemoveLayer": false,
  "isConfirmRemovePositions": false,
  "mosaic": Object {
    "id": "1000",
    "layers": Array [
      Object {
        "dimension": 3,
        "positions": Array [],
        "visible": true,
      },
    ],
    "lowLayersVisibility": true,
    "selectedLayer": 0,
  },
  "selectedPosition": "1",
}
`;

exports[`check setMosaic 1`] = `
Object {
  "dimensions": Array [
    1,
    2,
    3,
    4,
    5,
    6,
    8,
    9,
    10,
    12,
  ],
  "indexLayerToRemove": undefined,
  "isConfirmRemoveLayer": false,
  "isConfirmRemovePositions": false,
  "mosaic": Object {
    "id": "1",
    "layers": Array [
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "4",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "3",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "2",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "1",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
        ],
        "visible": true,
      },
      Object {
        "dimension": 4,
        "positions": Array [
          Object {
            "h": 2,
            "i": "6",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 0,
            "y": 1,
          },
          Object {
            "h": 2,
            "i": "5",
            "maxH": undefined,
            "maxW": undefined,
            "role": 0,
            "w": 2,
            "x": 2,
            "y": 1,
          },
        ],
        "visible": true,
      },
    ],
    "lowLayersVisibility": true,
    "selectedLayer": 2,
  },
  "selectedPosition": "",
}
`;
